scripts/gnomad-json.py

```diff
@@ -0,0 +1,177 @@
+import pandas as pd
+import json
+from statsmodels.stats import proportion
+
+def cyclical_variations(motifs):
+    # builds the set of all possible motifs with a given motif
+    # Motifs with N in them are all permuted with combination of all nucleotides
+    nucs = ['A', 'C', 'G', 'T']
+    variations = []
+    for motif in motifs:
+        if 'N' in motif:
+            for n in nucs:
+                variations += cyclical_variations([motif.replace('N', n)])
+        variations += [motif[-i:]+motif[:-i] for i in range(len(motif))]
+    return variations
+
+
+def binomial_ci(x, n, confidence=0.95):
+    """
+    Calculates the confidence interval for a binomial proportion.
+
+    Args:
+        x: Number of successes
+        n: Number of trials
+        confidence: Confidence level (e.g., 0.95 for a 95% CI)
+
+    Returns:
+        tuple: Lower and upper bounds of the confidence interval
+    """
+    
+    lower, upper = proportion.proportion_confint(x, n, alpha=1-confidence, method='binom_test')
+    return lower, upper
+
+def calc_plotdata(sub_df):
+    sorted_poplabels = ["amr", "afr", "ami", "asj", "nfe", "fin", "mid", "sas", "eas", "oth"]
+    population_labels = {
+        "amr": "Admixed American",
+        "afr": "African/African American",
+        "asj": "Ashkenazi Jewish",
+        "ami": "Amish",
+        "eas": "East Asian",
+        "fin": "Finnish",
+        "nfe": "European (non Finnish)",
+        "mid": "Middle Eastern",
+        "sas": "South Asian",
+        "oth": "Others",
+    }
+    labels = []
+    values = []
+    counts = []
+    xconf_lowerbound = []
+    xconf_upperbound = []
+    for population in sorted_poplabels:
+        labels.append(f'{population_labels[population]}')
+        count = len(sub_df[sub_df['Population']==population])
+        counts.append(count)
+        if True in sub_df[sub_df['Population']==population]['Pathogenic'].unique():
+            pcount = sub_df[sub_df['Population']==population]['Pathogenic'].value_counts()[True]
+            values.append((pcount/count)*100)
+            conf_lower, conf_upper = binomial_ci(pcount, count)
+        else:
+            values.append(0)
+            conf_lower, conf_upper = binomial_ci(0, count, 0.95)
+        xconf_lowerbound.append(conf_lower*100)
+        xconf_upperbound.append(conf_upper*100)
+    
+    return [values, labels, counts, xconf_lowerbound, xconf_upperbound]
+
+json_txt = ""
+with open('../data/STRchive-loci.json') as fh:
+    for line in fh: json_txt += line.strip()
+strchive_info = json.loads(json_txt)
+
+df = pd.read_csv('../data/gnomAD_STR_genotypes__including_all_age_and_pcr_info__2023_06_28.tsv.gz', sep='\t')
```

Comment:
I'd recommend making this filename more succinct, and also putting it in a separate folder like `/raw` to distinguish it from the top level data files which are meant to be used by people using the data directly.

Reply:
Related, where did this file come from? Is it directly from the gnomAD website?

Reply:
@laurelhiatt had shared the gnomAD TSV file with me. It is probably an earlier version of gnomAD STR genotypes file.

Reply:
> I'd recommend making this filename more succinct, and also putting it in a separate folder like `/raw` to distinguish it from the top level data files which are meant to be used by people using the data directly.

Just to confirm the gnomad TSV file, right? Should I be relocating to /raw or /data/raw?

Reply:
Renaming the file to gnomAD_STRgenotypes_20230628.tsv.gz and moving it to /data/raw